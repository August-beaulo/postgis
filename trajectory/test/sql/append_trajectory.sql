--
-- check_postgis_installed
--

\c trjregression

SET client_min_messages TO warning;

-- add TAXI
SELECT trajectory.CreateTrajectory('taxi', 'B129',  4326, 0.00001);

-- append 5 points
SELECT trajectory.AppendTrajectory('taxi', 'B129', '2015-10-10 8:10:00', ST_GeomFromText('POINT(119.4 39.4)', 4326));
SELECT trajectory.AppendTrajectory('taxi', 'B129', '2015-10-10 8:15:00',  ST_Transform(ST_GeomFromText('POIN    T(119.4 39.4)', 4326),2249));
SELECT trajectory.AppendTrajectory('taxi', 'B129', '2015-10-10 8:20:00', row(LOCALTIMESTAMP(2), ST_Point(119.4, 39.4))::trajectory.TPoint);
SELECT trajectory.AppendTrajectory('taxi', 'B129', '2015-10-10 8:25:00', row(LOCALTIMESTAMP(2), '(119.4, 39.4)'::point)::trajectory.TPoint);
SELECT trajectory.AppendTrajectory('taxi', 'B129', '2015-10-10 8:30:00', row(LOCALTIMESTAMP(2), '(119.3, 39.3)'::point)::trajectory.TPoint);

-- query them
SELECT M.poolname, M.trjname, T.time, ST_AsText(T.position)
    FROM trajectory M, taxi T WHERE M.id = T. id AND M.poolname = 'taxi' AND trjname = 'B129' ORDER BY T.time;

-- drop it
SELECT trajectory.DeleteTrajectory('taxi', 'B129');
