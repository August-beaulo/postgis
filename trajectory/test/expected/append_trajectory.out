--
-- check_postgis_installed
--
\c trjregression
SET client_min_messages TO warning;
-- add TAXI
SELECT trajectory.CreateTrajectory('taxi', 'B129',  4326, 0.00001);
 createtrajectory 
------------------
               12
(1 row)

-- append 5 points
SELECT trajectory.AppendTrajectory('taxi', 'B129', '2015-10-10 8:10:00', ST_GeomFromText('POINT(119.4 39.4)', 4326));
 appendtrajectory 
------------------
 t
(1 row)

SELECT trajectory.AppendTrajectory('taxi', 'B129', '2015-10-10 8:15:00',  ST_Transform(ST_GeomFromText('POIN    T(119.4 39.4)', 4326),2249));
ERROR:  parse error - invalid geometry (lwgeom_pg.c:90)
HINT:  "PO" <-- parse error at position 2 within geometry
SELECT trajectory.AppendTrajectory('taxi', 'B129', '2015-10-10 8:20:00', row(LOCALTIMESTAMP(2), ST_Point(119.4, 39.4))::trajectory.TPoint);
ERROR:  function trajectory.appendtrajectory(unknown, unknown, unknown, tpoint) does not exist
LINE 1: SELECT trajectory.AppendTrajectory('taxi', 'B129', '2015-10-...
               ^
HINT:  No function matches the given name and argument types. You may need to add explicit type casts.
SELECT trajectory.AppendTrajectory('taxi', 'B129', '2015-10-10 8:25:00', row(LOCALTIMESTAMP(2), '(119.4, 39.4)'::point)::trajectory.TPoint);
ERROR:  function trajectory.appendtrajectory(unknown, unknown, unknown, tpoint) does not exist
LINE 1: SELECT trajectory.AppendTrajectory('taxi', 'B129', '2015-10-...
               ^
HINT:  No function matches the given name and argument types. You may need to add explicit type casts.
SELECT trajectory.AppendTrajectory('taxi', 'B129', '2015-10-10 8:30:00', row(LOCALTIMESTAMP(2), '(119.3, 39.3)'::point)::trajectory.TPoint);
ERROR:  function trajectory.appendtrajectory(unknown, unknown, unknown, tpoint) does not exist
LINE 1: SELECT trajectory.AppendTrajectory('taxi', 'B129', '2015-10-...
               ^
HINT:  No function matches the given name and argument types. You may need to add explicit type casts.
-- query them
SELECT M.poolname, M.trjname, T.time, ST_AsText(T.position)
    FROM trajectory M, taxi T WHERE M.id = T. id AND M.poolname = 'taxi' AND trjname = 'B129' ORDER BY T.time;
 poolname | trjname |           time           |     st_astext     
----------+---------+--------------------------+-------------------
 taxi     | B129    | Sat Oct 10 08:10:00 2015 | POINT(119.4 39.4)
(1 row)

-- drop it
SELECT trajectory.DeleteTrajectory('taxi', 'B129');
                        deletetrajectory                        
----------------------------------------------------------------
 Trajectory taxi (B129) (with id = 12) is removed successfully.
(1 row)

